common/metadata.go

```diff
@@ -0,0 +1,156 @@
+	Precision *int `json:"precision,omitempty"`
+	// Scale is the number of digits to the right of the decimal point (for decimal types).
+	Scale *int `json:"scale,omitempty"`
+	// Required indicates if the field is required.
+	Required bool `json:"required,omitempty"`
+	// Unique indicates if the field must be unique across all records.
+	Unique bool `json:"unique,omitempty"`
+	// Indexed indicates if the field should be indexed for faster search.
+	Indexed bool `json:"indexed,omitempty"`
+	// DefaultValue is the default value for the field (if any).
+	DefaultValue string `json:"defaultValue,omitempty"`
+	// Min is the minimum value for numeric fields (if any).
+	Min *float64 `json:"min,omitempty"`
+	// Max is the maximum value for numeric fields (if any).
+	Max *float64 `json:"max,omitempty"`
+	// Pattern is a regex pattern that the field value must match (if any).
+	Pattern string `json:"pattern,omitempty"`
+	// Values is a list of allowed values for enum fields (if any).
+	Values []string `json:"values,omitempty"`
+	// ValuesRestricted indicates if the field value must be limited to what's in Values.
+	ValuesRestricted bool `json:"valuesRestricted,omitempty"`
+	// Association defines association/relationship information for the field (if any).
+	Association *AssociationDefinition `json:"association,omitempty"`
+}
+
+// AssociationDefinition defines relationship information for a field
+// to another object. Note that not all providers will support all
+// aspects of the association. This is a best-effort attempt to create
+// a common schema for associations across providers.
+//
+// In the event that a provider doesn't support a particular aspect of
+// the association, and assuming that the field has a value, then the
+// value should be ignored, and a warning should be added to the
+// UpsertMetadataResult.
+type AssociationDefinition struct {
+	// Kind is the high-level association variety (e.g., 'foreignKey', 'lookup', 'ref').
+	// The provider determines the exact behavior.
+	Kind string `json:"kind"`
+	// TargetObject is the name of the referenced/parent object.
+	TargetObject string `json:"targetObject"`
+	// TargetField is the name of the referenced field on the target object.
+	// Defaults to the target's primary key when omitted.
+	TargetField string `json:"targetField,omitempty"`
+	// Relationship cardinality from the referencing field's perspective (e.g., 'many-to-one', 'one-to-one').
+	Cardinality string `json:"cardinality,omitempty"`
+	// OnDelete defines the behavior upon foreign object deletion, where supported.
+	// E.g., 'cascade', 'restrict', 'setNull'.
+	OnDelete string `json:"onDelete,omitempty"`
+	// Required means that, if true, a referenced record must exist (i.e., NOT NULL foreign key).
+	Required bool `json:"required,omitempty"`
+	// InverseName is an optional inverse relationship/property name exposed on the target object.
+	InverseName string `json:"inverseName,omitempty"`
```

Comment:
I'm a bit confused by what this is still, is this like a field on the parent object that gets updated to reference the child object? If so, maybe "ReverseLookupFieldName"?

Reply:
I believe if you think of associations as a DAG, then as we're pointing at some foreign object, this sets up the foreign object to point back to us (and specifies the name). So it turns it from a 1-way association to a 2-way association.

Reply:
got it, then I would name it "ReverseLookupFieldName" for clarity.